tests/deploy/e2e/test_autostarted.py

```diff
@@ -315,3 +360,779 @@ async def test_autostart_clear_agent_venv_on_delete(
+""",
+        autostd=True,
+    )
+
+    # Now, let's deploy some resources
+    version, res, status = await snippetcompiler.do_export_and_deploy(include_status=True)
+    result = await client.release_version(environment, version, push=False)
+    assert result.code == 200
+
+    async def are_resources_being_deployed() -> bool:
+        result = await client.resource_list(environment, deploy_summary=True)
+        assert result.code == 200
+        summary = result.result["metadata"]["deploy_summary"]
+        deploying = summary["by_state"]["deploying"]
+        return deploying == 1
+
+    # Wait for at least one resource to be in deploying
+    await retry_limited(are_resources_being_deployed, timeout=5)
+
+    # Let's check the agent table and check that agent1 is present and not paused
+    await assert_is_paused(client, environment, {const.AGENT_SCHEDULER_ID: False, "agent1": False})
+
+    # Make sure the children of the scheduler are consistent
+    await wait_for_consistent_children(
+        current_pid=current_pid,
+        should_scheduler_be_defined=True,
+        should_fork_server_be_defined=True,
+        executor_to_be_defined=1,
+    )
+
+    # Wait for something to be deployed
+    try:
+        await clienthelper.wait_for_deployed(timeout=5)
+        assert not should_time_out, f"This was supposed to time out with a deployment sleep set to {time_to_sleep}!"
+    except (asyncio.TimeoutError, AssertionError):
+        result = await client.list_agents(tid=environment)
+        assert result.code == 200
+        assert should_time_out, f"This wasn't supposed to time out with a deployment sleep set to {time_to_sleep}!"
+    finally:
+        # Retrieve the current processes, we should have more processes than `start_children`
+        await wait_for_consistent_children(
+            current_pid=current_pid,
+            should_scheduler_be_defined=True,
+            should_fork_server_be_defined=True,
+            executor_to_be_defined=1,
+        )
+
+    state_after_deployment = construct_scheduler_children(current_pid)
+    for child in state_after_deployment.children:
+        assert child.is_running()
+
+    # The number of resources for this version should match
+    result = await client.list_versions(tid=environment)
+    assert result.code == 200
+    assert len(result.result["versions"]) == 1
+    assert result.result["versions"][0]["total"] == 1
+
+    # Let's check the agent table and check that agent1 is present and not paused
+    await assert_is_paused(client, environment, {const.AGENT_SCHEDULER_ID: False, "agent1": False})
+
+    # Now let's halt the environment
+    result = await client.halt_environment(tid=environment)
+    assert result.code == 200
+
+    snippetcompiler.setup_for_snippet(
+        f"""
+    import minimalwaitingmodule
+
+    a = minimalwaitingmodule::Sleep(name="test_sleep", agent="agent1", time_to_sleep={time_to_sleep})
+    """,
```

Comment:
Very minor detail the model could be factorized and reused from L546

Reply:
Would you like to change this to a function? If that's the goal, I think the test case will be harder to read

Reply:
No I meant this string into a variable, but this is fine either way really.